File: domain/src/main/java/com/beat/domain/promotion/repository/PromotionRepository.java
package com.beat.domain.promotion.repository;

import java.util.List;
import java.util.Optional;

import com.beat.domain.promotion.domain.CarouselNumber;
import com.beat.domain.promotion.domain.Promotion;

public interface PromotionRepository {

	List<Promotion> findAll();

	Optional<Promotion> findById(Long promotionId);

	Promotion save(Promotion promotion);

	List<Promotion> saveAll(List<Promotion> promotions);

	void deleteByPromotionIds(List<Long> promotionIds);

	void deleteByPerformanceId(Long performanceId);

	Optional<Promotion> findByCarouselNumber(CarouselNumber carouselNumber);
}

File: domain/src/main/kotlin/com/beat/domain/promotion/repository/PromotionRepository.kt
package com.beat.domain.promotion.repository

import com.beat.domain.promotion.domain.CarouselNumber
import com.beat.domain.promotion.domain.Promotion
import java.util.*

@JvmSuppressWildcards
interface PromotionRepository {
    fun findAll(): List<Promotion>

    fun findById(promotionId: Long?): Optional<Promotion>

    fun save(promotion: Promotion): Promotion

    fun saveAll(promotions: List<Promotion>): List<Promotion>

    fun deleteByPromotionIds(promotionIds: List<Long>)

    fun deleteByPerformanceId(performanceId: Long?)

    fun findByCarouselNumber(carouselNumber: CarouselNumber): Optional<Promotion>
}
